designs/images/cluster-controller-container-components.puml

```diff
@@ -0,0 +1,59 @@
+
+Person(user, "EKS-A user", "Manages EKS-A clusters")
+Container(kubeApiServer, "Kubernetes API server", "Go", "Provides read/write access to the kubernetes API objects stored in the etcd server and allows to subscribe to events when they change")
+Container_Ext(workloadKubeApiServer, "Workload Cluster Kubernetes API server", "Go", "Provides read/write access to the kubernetes API objects stored in the etcd server")
+Container_Ext(ociRegistry, "OCI registry", "Public ECR or other", "OCI compliant registry to store container images and Helm charts")
+
+Container_Boundary(clusteController, "Cluster Controller") {
+    AddElementTag("port", $fontColor="#ffffbf", $borderColor="#ffffbf")
+    AddElementTag("busines logic", $fontColor="#01ffff", $borderColor="#01ffff")
+
+    Component(clusterReconciler, "Cluster Reconciler", "K8s controller-runtime Reconciler", "Validates input and orchestrates Cluster reconcile requests", $tags="busines logic")
+    Component(vsphereHandler, "vSphere Cluster Reconciler", "ProviderClusterReconciler", "Reconciles a vSphere Cluster to the desired state", $tags="busines logic")
+    Component(snowHandler, "Snow Cluster Reconciler", "ProviderClusterReconciler", "Reconciles a Snow Cluster to the desired state", $tags="busines logic")
+
+    Component(cilimHandler, "Cilium Reconciler", "", "Reonciles the Cilium CNI installation to the desired state", $tags="busines logic")
+
+    Component(client, "Kubernetes API client", "controller-runtime client.Client", "Kubernetes client provided by controller-runtime", $tags="port")
+    Component(remoteClient, "Remote Kubernetes API client", "controller-runtime client.Client", "Kubernetes client for external cluster authenticated with kubeconfig", $tags="port")
+
+    Component(helm, "Helm client", "Helm cli binary [Go]", "Client to interact with Helm charts", $tags="port")
+
+    Rel(vsphereHandler, cilimHandler, "Delegates")
+    Rel(snowHandler, cilimHandler, "Delegates")
+
+    Rel(clusterReconciler, vsphereHandler, "Delegates")
+    Rel(clusterReconciler, snowHandler, "Delegates")
+
+    Rel(vsphereHandler, client, "Reads/writes CAPI objects through")
+    Rel(snowHandler, client, "Reads/writes CAPI objects through")
+    Rel(clusterReconciler, client, "Reads objects through")
+    Rel(cilimHandler, remoteClient, "Reads/writes Cilium DS/deployment and CRDs")
+
+    Rel(cilimHandler, helm, "Generates Cilium resources in yaml")
+}
+
+Container(capiControllers, "CAPI controllers", "Go", "Reoncile workload clusters")
+Rel(snowHandler, capiControllers, "Creates/updates snow clusters", $tags="indirect relantionship")
+Rel(vsphereHandler, capiControllers, "Creates/updates vSphere clusters", $tags="indirect relantionship")
+Rel(capiControllers, cilimHandler, "Marks control plane as ready", $tags="indirect relantionship")
+Rel(capiControllers, workloadKubeApiServer, "Creates", $tags="indirect relantionship")
```

Comment:
For the purpose of this diagram, I think you can drop arrows between components/containers _outside_ the boundary. These arrows tend to add clutter and detract from communicating the relationship of the components in the subject container.

(not to be confused with a user of the system and how they interact with it).

Reply:
Fair